index/postings_test.go

```diff
@@ -61,39 +61,125 @@ func TestMemPostings_ensureOrder(t *testing.T) {
 	}
 }
 
+func toPostings(refs ...uint64) Postings {
+	return newListPostings(refs)
+}
+
 func TestIntersect(t *testing.T) {
+	a := toPostings(1, 2, 3)
+	b := toPostings(2, 3, 4)
+
 	var cases = []struct {
-		a, b []uint64
-		res  []uint64
+		in []Postings
+
+		res Postings
 	}{
 		{
-			a:   []uint64{1, 2, 3, 4, 5},
-			b:   []uint64{6, 7, 8, 9, 10},
-			res: nil,
+			in:  []Postings{},
+			res: EmptyPostings(),
 		},
 		{
-			a:   []uint64{1, 2, 3, 4, 5},
-			b:   []uint64{4, 5, 6, 7, 8},
-			res: []uint64{4, 5},
+			in:  []Postings{a, b, EmptyPostings()},
+			res: EmptyPostings(),
 		},
 		{
-			a:   []uint64{1, 2, 3, 4, 9, 10},
-			b:   []uint64{1, 4, 5, 6, 7, 8, 10, 11},
-			res: []uint64{1, 4, 10},
-		}, {
-			a:   []uint64{1},
-			b:   []uint64{0, 1},
-			res: []uint64{1},
+			in:  []Postings{b, a, EmptyPostings()},
+			res: EmptyPostings(),
+		},
+		{
+			in:  []Postings{EmptyPostings(), b, a},
+			res: EmptyPostings(),
+		},
+		{
+			in:  []Postings{EmptyPostings(), a, b},
+			res: EmptyPostings(),
+		},
+		{
+			in:  []Postings{a, EmptyPostings(), b},
+			res: EmptyPostings(),
+		},
+		{
+			in:  []Postings{b, EmptyPostings(), a},
+			res: EmptyPostings(),
+		},
+		{
+			in:  []Postings{b, EmptyPostings(), a, a, b, a, a, a},
+			res: EmptyPostings(),
+		},
+		{
+			in: []Postings{
+				toPostings(1, 2, 3, 4, 5),
+				toPostings(6, 7, 8, 9, 10),
+			},
+			res: toPostings(),
+		},
+		{
+			in: []Postings{
+				toPostings(1, 2, 3, 4, 5),
+				toPostings(4, 5, 6, 7, 8),
+			},
+			res: toPostings(4, 5),
+		},
+		{
+			in: []Postings{
+				toPostings(1, 2, 3, 4, 9, 10),
+				toPostings(1, 4, 5, 6, 7, 8, 10, 11),
+			},
+			res: toPostings(1, 4, 10),
+		},
+		{
+			in: []Postings{
+				toPostings(1),
+				toPostings(0, 1),
+			},
+			res: toPostings(1),
+		},
+		{
+			in: []Postings{
+				toPostings(1),
+			},
+			res: toPostings(1),
+		},
+		{
+			in: []Postings{
+				toPostings(1),
+				toPostings(),
+			},
+			res: toPostings(),
+		},
+		{
+			in: []Postings{
+				toPostings(),
+				toPostings(),
+			},
+			res: toPostings(),
 		},
 	}
 
 	for _, c := range cases {
-		a := newListPostings(c.a)
-		b := newListPostings(c.b)
+		t.Run("", func(t *testing.T) {
```

Comment:
maybe add some indication to know when a test fails which case failed.
```
map[string]struct {
.
.

for t, c := range cases {
		t.Run(t, func(t *testing.T) {
```

Reply:
not needed. Passing `""` cause go test to put a number, so exactly what you propose.

Reply:
yes that is right but when you have many cases it is kind of difficult to count which one failed exactly.

Reply:
Disagree, let's not obfuscate the tests more -> everyone can count

Reply:
that is why I added the title in:
https://github.com/prometheus/tsdb/pull/559/files

Reply:
Yea, I agree there is a value for this, but I am afraid I won't have time to do this, this week ): 

I would suggesting merging this PR with just indexing (numbers), then we can work on making it more verbose later on. What do you think?

Reply:
sure I don't mind can always add it later if it gets annoying to count :smile: